File: solutions/009.palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0 ||(x%10==0&&x!=0)){
            return false;
        }else{
            int temp = 0;
            while(temp < x){
                temp = temp*10 + x%10;
                x = x/10;
            }
            if(x==temp || temp/10 ==x ){
                return true;
            }else{
                return false;
            }
        }
        
    }
}

File: solutions/009.palindrome-number/palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if(x<0)
        return false
    let sourceString = String(x)
    let length = sourceString.length
    let firstPart= '', lastPart =''
    if(length%2 === 0){
        for(let i =0; i<length/2;i++){
            if(sourceString[i]!==sourceString[length-1-i])
                return false
        }
    }else{
        for(let i = 0;i<(length-1)/2;i++){
            if(sourceString[i] !== sourceString[length-1-i])
                return false
        }
    }
    return true
    
    
    
    
};